lib/core.js

```diff
@@ -300,13 +300,21 @@ interface IteratorResult<T> {
     value?: T;
 }
 
-interface Iterator<T> {
-    next(): IteratorResult<T>;
-    @@iterator(): Iterator<T>;
+interface $Iterator<Y,R,N> {
+    @@iterator(): $Iterator<Y,R,N>;
+    next(value?: N): IteratorResult<Y|R>;
 }
+type Iterator<T> = $Iterator<T,void,void>;
 
-interface Iterable<T> {
-    @@iterator(): Iterator<T>;
+interface $Iterable<Y,R,N> {
```

Comment:
I have no objection. I can add a commit here, but it seems like an easy enough change to make in your internal merge, too, if you want to make the change yourself. (I'm not sure what the internal merge looks like, of course, so this could be wrong-headed of me.)